gravitee-am-ui/src/app/environment/environment.component.ts

```diff
@@ -51,7 +51,8 @@ export class EnvironmentComponent implements OnInit {
     // redirect user to the first domain, if any.
     this.domainService.list().subscribe((response) => {
       if (response.data && response.data.length > 0) {
-        this.router.navigate(['/environments', this.currentEnvironment.name, 'domains', response.data[0].hrid]);
+        const envHrid = this.currentEnvironment.hrids.length > 0 ? this.currentEnvironment.hrids[0] : this.currentEnvironment.id;
```

Comment:
currentEnvironment.hrids[0] is used in html too. 

Please extract the method and use it there too.